frontend/app/components/activity-new.js

```diff
@@ -34,14 +34,31 @@ export default Component.extend(EKMixin, {
     context.sendAction("done", true);
   },
 
+  addMissingMonthWarning(newActivity) {
+    if (
+      !newActivity.monthFrom ||
+      (!newActivity.monthTo && newActivity.yearTo)
+    ) {
```

Comment:
Put this check into the record as a function... Maybe even into daterange-model.js and give it a smart name.
That way it is reusable and only bound to the model.